### PR TITLE
[wip] Add builder hyperv-vhd

### PR DESCRIPTION
**Pull request by BSick7**
NOTE: This is a work-in-progress.  I would like early feedback.

## Overview

Packer currently only supports building from an iso.  This builder utilizes a source vhd rather than iso to create a hyperv image.  This is intended to be similar to vmware vmx or virtualbox ovf.

## Alterations

In order to reuse functionality from the iso builder, I have abstracted things from the iso builder.
- guest additions
- sizing (ram + disk)

### CODE REVIEW

**Comment by taliesins** [comment]
I am going to merge #4944 this weekend. Is there a good reason I should not? If I do there may be wasted effort on this PR. Could you have a quick look at it.

**Comment by BSick7** [comment]
Sorry I missed this potentially duplicate work.
I'll post some comments on that thread.

File: builder/hyperv/vhd/builder.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		&hypervcommon.StepExportVm{
+			OutputDir:      b.config.OutputDir,
+			SkipCompaction: b.config.SkipCompaction,
+		},
+
+		// the clean up actions for each step will be executed reverse order
+	}
+
+	// Run the steps.
+	if b.config.PackerDebug {
```
Comment (rickard-von-essen) [inline builder/hyperv/vhd/builder.go]: This is the wrong way of doing it. Check how it's done in [virtualbox-iso](https://github.com/hashicorp/packer/blob/master/builder/virtualbox/iso/builder.go#L296).

Comment (BSick7) [reply builder/hyperv/vhd/builder.go]: Good feedback. Thanks!

This was lifted from `hyperv-iso`.
Think it useful to update that builder as well?

Comment (rickard-von-essen) [reply builder/hyperv/vhd/builder.go]: That was just fixed there.

**Comment by SwampDragons** [comment]
Looks like this has been addressed on in a separate PR; closing this.

**Pull request closed without merging**